File: app/src/main/java/dev/ragnarok/fenrir/mvp/view/IAudioPlaylistsView.java
package dev.ragnarok.fenrir.mvp.view;

import java.util.List;

import dev.ragnarok.fenrir.model.AudioPlaylist;
import dev.ragnarok.fenrir.mvp.core.IMvpView;
import dev.ragnarok.fenrir.mvp.view.base.IAccountDependencyView;


public interface IAudioPlaylistsView extends IAccountDependencyView, IMvpView, IErrorView {
    void displayData(List<AudioPlaylist> pages);

    void notifyDataSetChanged();

    void notifyItemRemoved(int position);

    void notifyDataAdded(int position, int count);

    void showRefreshing(boolean refreshing);

    void doAddAudios(int accountId);

    void showHelper();
}

File: app/src/main/java/dev/ragnarok/fenrir/mvp/view/IAudioPlaylistsView.kt
package dev.ragnarok.fenrir.mvp.view

import dev.ragnarok.fenrir.model.AudioPlaylist
import dev.ragnarok.fenrir.mvp.core.IMvpView
import dev.ragnarok.fenrir.mvp.view.base.IAccountDependencyView

interface IAudioPlaylistsView : IAccountDependencyView, IMvpView, IErrorView {
    fun displayData(pages: List<AudioPlaylist>)
    fun notifyDataSetChanged()
    fun notifyItemRemoved(position: Int)
    fun notifyDataAdded(position: Int, count: Int)
    fun showRefreshing(refreshing: Boolean)
    fun doAddAudios(accountId: Int)
    fun showHelper()
}